src/software/simulation/simulator.cpp

```diff
@@ -84,27 +84,49 @@ void Simulator::setRobotPrimitives(
         return;
     }
 
-    SimulatorBallSingleton::setSimulatorBall(simulator_ball);
     for (const auto& primitive_ptr : *primitives)
     {
         primitive_params_t primitive_params = getPrimitiveParams(primitive_ptr);
         unsigned int primitive_index        = getPrimitiveIndex(primitive_ptr);
 
-        auto simulator_robots_iter =
-            std::find_if(simulator_robots.begin(), simulator_robots.end(),
-                         [&primitive_ptr](const auto& robot_world_pair) {
-                             return robot_world_pair.first->getRobotId() ==
-                                    primitive_ptr->getRobotId();
-                         });
+        setRobotPrimitive(primitive_ptr->getRobotId(), primitive_index, primitive_params,
+                          simulator_robots, simulator_ball);
+    }
+}
 
-        if (simulator_robots_iter != simulator_robots.end())
-        {
-            auto simulator_robot = (*simulator_robots_iter).first;
-            auto firmware_world  = (*simulator_robots_iter).second;
-            SimulatorRobotSingleton::setSimulatorRobot(simulator_robot);
-            SimulatorRobotSingleton::startNewPrimitiveOnCurrentSimulatorRobot(
-                firmware_world, primitive_index, primitive_params);
-        }
+void Simulator::setYellowRobotPrimitive(RobotId id, unsigned int primitive_index,
+                                        const primitive_params_t& params)
+{
+    setRobotPrimitive(id, primitive_index, params, yellow_simulator_robots,
+                      simulator_ball);
+}
+
+void Simulator::setBlueRobotPrimitive(RobotId id, unsigned int primitive_index,
```

Comment:
It used to be an enum but I changed it back because this is what the `PrimitiveManager` uses, and the enum is defined in proto which I didn't want to spread everywhere.

Will update the comment.

Reply:
is this the primitive type? Will it go away when primitive protos become a oneof?

Reply:
correct